app/core/app_state.py

```diff
@@ -12,13 +12,14 @@
 from .database import DatabaseManager
 from .edge_inference import EdgeInferenceManager
 from .file_paths import DEFAULT_EDGE_CONFIG_PATH
-from .utils import safe_call_sdk
+from .utils import TimestampedTTLCache, safe_call_sdk
 
 logger = logging.getLogger(__name__)
 
 MAX_SDK_INSTANCES_CACHE_SIZE = 1000
 MAX_DETECTOR_IDS_TTL_CACHE_SIZE = 1000
-TTL_TIME = 3600  # 1 hour
+TTL_TIME = 600  # 10 minutes
+STALE_METADATA_THRESHOLD = 30  # 30 seconds
```

Comment:
style nit: it can be nice to define time values, like `TEN_MINUTES = 60 * 10` and then to use these constants directly. Like `if cached_value_age > THIRTY_SECONDS:` below.
Separately, its a good idea to include units in variable names. So instead of `STALE_METADATA_THRESHOLD`, `STALE_METADATA_THRESHOLD_SEC`